src/dns.rs

```diff
@@ -49,12 +49,12 @@ impl Dns {
         addrs.to_ip_addrs(self)
     }
 
-    pub(crate) fn reverse(&self, addr: IpAddr) -> &str {
+    pub(crate) fn reverse(&self, addr: IpAddr) -> String {
```

Comment:
Perhaps we can make the internal version fallible and leverage that within the tracing calls, but keep the public API as is. So this would return `Option<&str>` and then we just `.expect(..)` on it for now. We can consider making the public ones fallible too in a later release.

This was purely added for debugging convenience as its easier to look at `&str` hostnames instead of IP addresses.

Reply:
Seems like the best solution for now.